torch/csrc/cuda/Graph.cpp

```diff
@@ -30,9 +31,30 @@ void THCPGraph_init(PyObject* module) {
       // docs aren't clear. But it works.
       .def(
           "capture_begin",
-          torch::wrap_pybind_function_no_gil(
-              &at::cuda::CUDAGraph::capture_begin),
-          py::arg("pool") = c10::cuda::MempoolId_t{0, 0})
+          [](::at::cuda::CUDAGraph& self,
+             c10::optional<c10::cuda::MempoolId_t> pool_opt,
+             std::string capture_error_mode) {
+            cudaStreamCaptureMode capture_mode;
+            c10::cuda::MempoolId_t pool = pool_opt.has_value()
+                ? pool_opt.value()
+                : c10::cuda::MempoolId_t{0, 0};
+            if (capture_error_mode == "global") {
+              capture_mode = cudaStreamCaptureModeGlobal;
+            } else if (capture_error_mode == "thread_local") {
+              capture_mode = cudaStreamCaptureModeThreadLocal;
+            } else if (capture_error_mode == "relaxed") {
+              capture_mode = cudaStreamCaptureModeRelaxed;
+            } else {
+              TORCH_INTERNAL_ASSERT(
+                  false,
+                  "Unknown capture error mode. Expected `global`, `thread_local`, or `relaxed`, got ",
+                  capture_error_mode);
+            }
+            return self.capture_begin(pool, capture_mode);
+          },
+          py::arg("pool") = (c10::optional<c10::cuda::MempoolId_t>)c10::nullopt,
```

Comment:
Why was this changed to a nullopt?
Also these args are always passed so no need to provide default for them?

Reply:
I was trying to clean up this call: https://github.com/pytorch/pytorch/pull/107407/files#diff-39e542d87359e7d5381d036cbcea9ec759fbe469578bcdc5693ce6cfab7f1a54L72-L77.

Reply:
But you still pass None here. So this default value will never be used?
And I guess our custom typecast is already properly translating the `None` that you pass in python into a `nullopt` here.

Reply:
Removed arg defaults.